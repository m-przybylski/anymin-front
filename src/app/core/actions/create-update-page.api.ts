// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';

export enum SessionUpdateApiActionTypes {
  CreateUpdateNameAndAvatar = '[create update session] Name and avatar',
}

export class CreateUpdateNameAndAvatarAction implements Action {
  public readonly type = SessionUpdateApiActionTypes.CreateUpdateNameAndAvatar;

  constructor(public payload: { name: string; avatarToken: string }) {}
}

export type SessionUpdateApiActionUnion = CreateUpdateNameAndAvatarAction;
