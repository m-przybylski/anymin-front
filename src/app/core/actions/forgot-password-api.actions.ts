import { Action } from '@ngrx/store';

export enum ForgotPasswordApiActionsTypes {
  SetMsisdnToken = '[API forgotPassword] SetMsisdnToken',
  DeleteMsisdnToken = '[API forgotPassword] DeleteMsisdnToken',
}

export class SetMsisdnToken implements Action {
  public readonly type = ForgotPasswordApiActionsTypes.SetMsisdnToken;

  constructor(public payload: string) {}
}

export class UnsetMsisdnTokenAction implements Action {
  public readonly type = ForgotPasswordApiActionsTypes.DeleteMsisdnToken;
}

export type ForgotPasswordApiActionsUnion = SetMsisdnToken | UnsetMsisdnTokenAction;
