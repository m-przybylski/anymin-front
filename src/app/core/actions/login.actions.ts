import { Action } from '@ngrx/store';
import { LoginCredentials, GetSessionWithAccount } from '@anymind-ng/api';

export enum AuthActionTypes {
  Login = '[auth] Login',
  LoginFromModal = '[auth] Login from modal',
  LoginFromModalSuccess = '[auth] Login from modal success',
  LoginError = '[auth] Login error',
  LoginSuccess = '[auth] Login success',
  Logout = '[auth] Logout',
  LogoutError = '[auth] Logout error',
  LogoutSuccess = '[auth] Logout success',
  LoginRedirect = '[auth] Login redirect',
  DashboardRedirect = '[auth] Dashboard redirect',
  LogoutRemote = '[auth] Logout by remote event',
  FirstLoginAfterRegistration = '[auth] First login and redirect after registration',
  UpdateFirstTimeLoginStatus = '[auth] First login and redirect update status',
}

export class LoginAction implements Action {
  public readonly type = AuthActionTypes.Login;

  constructor(public payload: LoginCredentials) {}
}

export class LoginModalAction implements Action {
  public readonly type = AuthActionTypes.LoginFromModal;

  constructor(public payload: LoginCredentials) {}
}

export class LoginErrorAction implements Action {
  public readonly type = AuthActionTypes.LoginError;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}
export class LoginModalSuccessAction implements Action {
  public readonly type = AuthActionTypes.LoginFromModalSuccess;
}

export class LoginSuccessAction implements Action {
  public readonly type = AuthActionTypes.LoginSuccess;

  constructor(public payload: GetSessionWithAccount) {}
}

export class LogoutAction implements Action {
  public readonly type = AuthActionTypes.Logout;
}
export class LogoutErrorAction implements Action {
  public readonly type = AuthActionTypes.LogoutError;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}
export class LogoutSuccessAction implements Action {
  public readonly type = AuthActionTypes.LogoutSuccess;
}
export class LoginRedirectAction implements Action {
  public readonly type = AuthActionTypes.LoginRedirect;
}
export class DashboardRedirectAction implements Action {
  public readonly type = AuthActionTypes.DashboardRedirect;

  constructor(public payload: GetSessionWithAccount) {}
}
export class LogoutRemoteAction implements Action {
  public readonly type = AuthActionTypes.LogoutRemote;
}
export class FirstLoginAfterRegistrationAction implements Action {
  public readonly type = AuthActionTypes.FirstLoginAfterRegistration;
}
export class UpdateFirstTimeLoginStatusAction implements Action {
  public readonly type = AuthActionTypes.UpdateFirstTimeLoginStatus;
}

export type AuthActionsUnion =
  | LoginAction
  | LoginErrorAction
  | LoginSuccessAction
  | LoginRedirectAction
  | LogoutAction
  | LogoutErrorAction
  | LogoutSuccessAction
  | DashboardRedirectAction
  | FirstLoginAfterRegistrationAction
  | UpdateFirstTimeLoginStatusAction
  | LoginModalAction
  | LoginModalSuccessAction
  | LogoutRemoteAction;
