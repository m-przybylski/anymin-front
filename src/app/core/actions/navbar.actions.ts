// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { UserTypeEnum } from '@platform/core/reducers/navbar.reducer';

export enum NavbarActionTypes {
  SetUserType = '[navbar] set user type',
  UpdateSessionAndUserType = '[navbar] set user type and fetch session',
  ToggleUserMenuVisibility = '[navbar] toggle user menu visibility',
  ToggleHelpMenuVisibility = '[navbar] toggle help menu visibility',
}

export class SetUserType implements Action {
  public readonly type = NavbarActionTypes.SetUserType;

  constructor(public payload: UserTypeEnum) {}
}

export class UpdateUserTypeAndSession implements Action {
  public readonly type = NavbarActionTypes.UpdateSessionAndUserType;

  constructor(public payload: UserTypeEnum) {}
}

export class ToggleUserMenuVisibility implements Action {
  public readonly type = NavbarActionTypes.ToggleUserMenuVisibility;
}

export class ToggleHelpMenuVisibility implements Action {
  public readonly type = NavbarActionTypes.ToggleHelpMenuVisibility;
}

export type NavbarActionsUnion =
  | SetUserType
  | UpdateUserTypeAndSession
  | ToggleUserMenuVisibility
  | ToggleHelpMenuVisibility;
