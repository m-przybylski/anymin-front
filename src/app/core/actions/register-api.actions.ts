import { Action } from '@ngrx/store';
import { GetSessionWithAccount } from '@anymind-ng/api';

export enum RegisterApiActionsTypes {
  RegisterError = '[API register] RegisterError',
  RegisterSuccess = '[API register] RegisterSuccess',
}

export class RegisterErrorAction implements Action {
  public readonly type = RegisterApiActionsTypes.RegisterError;

  constructor(public payload: any) {}
}

export class RegisterSuccessAction implements Action {
  public readonly type = RegisterApiActionsTypes.RegisterSuccess;

  constructor(public payload: GetSessionWithAccount) {}
}

export type RegisterAPIActionsUnion = RegisterErrorAction | RegisterSuccessAction;
