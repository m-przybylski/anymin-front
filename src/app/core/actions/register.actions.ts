import { Action } from '@ngrx/store';
import { GetSessionWithAccount, PostAccount } from '@anymind-ng/api';

export enum RegisterActionsTypes {
  Register = '[register] Register',
  RegisterByModal = '[register] Register by modal',
  RedirectToDashboardInvitations = '[register] Redirect to invitation list',
  RedirectToDashboardActivities = '[register] Register to activities',
}

export class RegisterAction implements Action {
  public readonly type = RegisterActionsTypes.Register;

  constructor(public payload: PostAccount) {}
}
export class RegisterByModalAction implements Action {
  public readonly type = RegisterActionsTypes.RegisterByModal;

  constructor(public payload: PostAccount) {}
}
export class RegisterRedirectToDashboardsInvitationsAction implements Action {
  public readonly type = RegisterActionsTypes.RedirectToDashboardInvitations;

  constructor(public payload: GetSessionWithAccount) {}
}

export class RegisterRedirectToDashboardsActivitiesAction implements Action {
  public readonly type = RegisterActionsTypes.RedirectToDashboardActivities;
}

export type RegisterActionsUnion =
  | RegisterAction
  | RegisterByModalAction
  | RegisterRedirectToDashboardsInvitationsAction
  | RegisterRedirectToDashboardsActivitiesAction;
