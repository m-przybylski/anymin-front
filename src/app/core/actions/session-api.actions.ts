// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { Account, GetSessionWithAccount } from '@anymind-ng/api';

export enum SessionWithAccountApiActionTypes {
  UpdateAccount = '[API Session with Account] update Account',
  FetchSessionError = '[API session] Fetch error',
  FetchSessionSuccess = '[API session] Fetch success',
  VerifyAccountByEmail = '[API session] Verify Account by Email',
  VerifyAccountByPin = '[API session] Verify Account By PIN',
}

export class UpdateAccountInSession implements Action {
  public readonly type = SessionWithAccountApiActionTypes.UpdateAccount;

  constructor(public payload: Account) {}
}

export class FetchSessionErrorAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.FetchSessionError;

  // tslint:disable-next-line:no-any
  constructor(public payload: any) {}
}
export class FetchSessionSuccessAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.FetchSessionSuccess;

  constructor(public payload: GetSessionWithAccount) {}
}

export class VerifyAccountByEmailAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.VerifyAccountByEmail;

  constructor(public payload: GetSessionWithAccount) {}
}

export class VerifyAccountByPinAction implements Action {
  public readonly type = SessionWithAccountApiActionTypes.VerifyAccountByPin;

  constructor(public payload: GetSessionWithAccount) {}
}

export type SessionAPIActionUnion =
  | UpdateAccountInSession
  | FetchSessionErrorAction
  | FetchSessionSuccessAction
  | VerifyAccountByPinAction
  | VerifyAccountByEmailAction;
