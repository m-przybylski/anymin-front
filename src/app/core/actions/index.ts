import * as AuthActions from './login.actions';
import * as SessionActions from './session.actions';
import * as NavbarActions from './navbar.actions';
import * as SessionUpdateApiActions from './create-update-page.api';
import * as SessionApiActions from './session-api.actions';

export { AuthActions, SessionActions, NavbarActions, SessionUpdateApiActions, SessionApiActions };
