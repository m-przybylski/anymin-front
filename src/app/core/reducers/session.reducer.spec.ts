import { IState, reducer } from './session.reducer';
import { SessionUpdateApiActions } from '../actions';

describe('balance.reducer', () => {
  const initialState: IState = {
    isFromBackend: false,
    isPending: false,
  };

  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toEqual(initialState);
    });
  });

  describe('CreateUpdateNameAndAvatarAction', () => {
    it('should update store even if no session', () => {
      const result = reducer(
        initialState,
        new SessionUpdateApiActions.CreateUpdateNameAndAvatarAction({ name: 'ArtuÅ›', avatarToken: 'ðŸ’©' }),
      );
      expect(result).toEqual(initialState as any);
    });
    it('should update store even if session exists', () => {
      const storeWithSession: any = {
        isFromBackend: true,
        isPending: false,
        session: {
          account: {
            id: 'c3d262b0-48d7-41b5-9aeb-9f59affdd0f3',
            msisdn: '+48607508372',
            email: 'maciej.przybylski@anymind.com',
            registeredAt: new Date('2018-08-09T10:00:35.039Z'),
            details: {
              nickname: 'ASD',
              avatar: 'fsafasdf',
            },
            isBlocked: false,
            hasPassword: true,
            isAnonymous: true,
          },
          session: {
            accountId: '123',
            apiKey: 'zxxx',
            city: 'sdf',
            country: 'aaa',
            ipAddress: '13123',
            isExpired: false,
            lastActivityAt: new Date('2018-11-09T10:00:35.039Z'),
            userAgent: 'aa',
          },
          isCompany: false,
          isExpert: false,
        },
      };
      const result = reducer(
        storeWithSession,
        new SessionUpdateApiActions.CreateUpdateNameAndAvatarAction({ name: 'ArtuÅ›', avatarToken: 'ðŸ’©' }),
      );
      expect(result).toMatchSnapshot();
    });
  });
});
