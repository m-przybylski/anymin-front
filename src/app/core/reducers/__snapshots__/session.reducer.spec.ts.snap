// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`session.reducer CreateUpdateNameAndAvatarAction should update store even if session exists 1`] = `
Object {
  "isFromBackend": true,
  "isPending": false,
  "session": Object {
    "account": Object {
      "details": Object {
        "avatar": "ðŸ’©",
        "nickname": "ArtuÅ›",
      },
      "email": "fgg@hfd.pl",
      "hasPassword": true,
      "id": "c3d262b0-48d7-41b5-9aeb-9f59affdd0f3",
      "isAnonymous": true,
      "isBlocked": false,
      "msisdn": "81927398127",
      "registeredAt": 2018-08-09T10:00:35.039Z,
    },
    "isCompany": false,
    "isExpert": false,
    "session": Object {
      "accountId": "123",
      "apiKey": "zxxx",
      "city": "sdf",
      "country": "aaa",
      "ipAddress": "13123",
      "isExpired": false,
      "lastActivityAt": 2018-11-09T10:00:35.039Z,
      "userAgent": "aa",
    },
  },
}
`;

exports[`session.reducer UpdateAccountInSession should update account in GetSessionWithAccount object 1`] = `
Object {
  "isFromBackend": true,
  "isPending": false,
  "session": Object {
    "account": Object {
      "countryISO": "PL",
      "currency": "PLN",
      "details": Object {
        "avatar": "6d3587a32aba453e8ad47199324f4c67",
        "clientId": "c3d262b0-48d7-41b5-9aeb-9f59affdd0f3",
        "nickname": "ðŸ¦„ðŸ¦„ðŸ¦„ðŸ¦„ðŸ¦„ðŸ¦„ðŸ¦„ðŸ¦„",
      },
      "email": "ðŸ¦„ðŸ¦„@ðŸ¦„ðŸ¦„.pl",
      "hasPassword": true,
      "id": "c3d262b0-48d7-41b5-9aeb-9f59affdd0f3",
      "isAnonymous": true,
      "isBlocked": false,
      "msisdn": "ðŸ¦„",
      "registeredAt": 2018-08-09T10:00:35.039Z,
      "vatRateType": "COMPANY_23",
    },
    "isCompany": false,
    "isExpert": false,
    "session": Object {
      "accountId": "123",
      "apiKey": "zxxx",
      "city": "sdf",
      "country": "aaa",
      "ipAddress": "13123",
      "isExpired": false,
      "lastActivityAt": 2018-11-09T10:00:35.039Z,
      "userAgent": "aa",
    },
  },
}
`;
